### ISSUE TITLE
Request for pdist API

### ISSUE THREAD

**Issue by WindQAQ** [tensorflow/addons#533]
**Describe the feature and the current behavior/state.**
`tfa.losses.metric_learning.pairwise_distance` computes pairwise 2-norm distance between each sample points. However, it is much slower than `torch.nn.functional.pdist` and even slower than `scipy.spatial.distance.pdist`. Though `torch` and `scipy` returns flatten upper triangular matrix only (the distance matrix is symmetric).

https://colab.research.google.com/drive/1MBpmpwlwZ_khCJLkIN0ztUzvuOwgfYvH

**Relevant information**
- Are you willing to contribute it (yes/no): Yes
- Are you willing to maintain it going forward? (yes/no): Yes
- Is there a relevant academic paper? (if so, where): no
- Is there already an implementation in another framework? (if so, where): Yes, in `scipy` and `torch`
- Was it part of tf.contrib? (if so, where): No

**Which API type would this fall under (layer, metric, optimizer, etc.)**
I have no idea where `pdist` would fall under. Probably a new folder?

**Who will benefit with this feature?**
Developers who need to compute distance/similarity matrix and researchers who focus on metric/embedding learning.

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
